### PR TITLE
Setup proper CMake for TSC

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
There are some incremental build issues to figure out but this mostly works.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+cmake_minimum_required(VERSION 3.15.1)
+
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL Darwin)
+    execute_process(
+        COMMAND xcrun --sdk macosx --show-sdk-path
+        OUTPUT_VARIABLE CMAKE_OSX_SYSROOT
+        OUTPUT_STRIP_TRAILING_WHITESPACE)
+    SET(CMAKE_Swift_FLAGS "${CMAKE_Swift_FLAGS} -sdk ${CMAKE_OSX_SYSROOT}")
+endif()
```
Comment (compnerd) [inline CMakeLists.txt]: Please dont modify `CMAKE_Swift_FLAGS`, its meant to be used by users only.  You can use `append_compile_options` with a generator expression to add the flags if you must.  (L18 has an example of the generator expression).

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+project(SwiftTSC LANGUAGES C Swift)
+ 
+set(SWIFT_VERSION 5)
```
Comment (compnerd) [inline CMakeLists.txt]: Why have a variable for this?  Why not just set it directly on L15?

Comment (aciidb0mb3r) [reply CMakeLists.txt]: It's also used in L18

File: CMakeLists.txt (hunk 1 above)
Comment (aciidb0mb3r) [reply CMakeLists.txt]: Hm, the generator expression doesn't work when CMake is trying to check for the Swift compiler. Is there any other way to fix that?

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci test

**Comment by aciidb0mb3r** [comment]
@swift-ci test

**Comment by aciidb0mb3r** [comment]
@swift-ci test

**Pull request merged**